src/MySqlConnector/MySqlClient/MySqlConnection.cs

```diff
@@ -20,6 +20,11 @@ public MySqlConnection()
 
 		public MySqlConnection(string connectionString) => ConnectionString = connectionString;
 
+		public MySqlConnection(MySqlConnectionStringBuilder connectionStringBuilder)
+		{
+			SetConnectionSettings(connectionStringBuilder);
```

Comment:
We need a private copy of this object, to avoid problems if the caller mutates it after passing it to the constructor.